main.go

```diff
@@ -561,6 +561,7 @@ func main() {
 	if err := app.Run(os.Args); err != nil {
 		log.Fatal(err)
 	}
+	defer db.Close()
```

Comment:
The reason we use `defer` is to ensure some action happens before we exit the current scope. So this should be defined right after we declare the `db` variable. This case is a bit tricky since `db` might be `nil`, so we could do something like this:
```go
var db *sql.DB
defer func() {
  if db != nil {
    db.Close()
  }
}()